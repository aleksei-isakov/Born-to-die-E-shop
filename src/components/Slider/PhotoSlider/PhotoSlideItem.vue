<template>
  <transition :name="transition">
    <slot></slot>
  </transition>
</template>

<script>
export default {
  name: 'PhotoSlideItem',

  props: {
    direction: {
      type: String,
      default: 'right',
      required: true
    }
  },

  computed: {
    transition() {
      return `slide-${this.direction}`;
    }
  }
};
</script>

<style scoped>
.slide-right-enter-active {
  animation: slideRightIn 0.5s;
}

.slide-right-leave-active {
  animation: slideRightOut 0.5s;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  width: 100%;
}

@keyframes slideRightIn {
  from {
    transform: translateX(100%);
  }
  to {
    transform: translateX(0);
  }
}

@keyframes slideRightOut {
  from {
    transform: translateX(0);
  }
  to {
    transform: translateX(-100%);
  }
}

.slide-left-enter-active {
  animation: slideLeftIn 0.5s;
}

.slide-left-leave-active {
  animation: slideLeftOut 0.5s;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  width: 100%;
}

@keyframes slideLeftIn {
  from {
    transform: translateX(-100%);
  }
  to {
    transform: translateX(0);
  }
}

@keyframes slideLeftOut {
  from {
    transform: translateX(0);
  }
  to {
    transform: translateX(100%);
  }
}
</style>
