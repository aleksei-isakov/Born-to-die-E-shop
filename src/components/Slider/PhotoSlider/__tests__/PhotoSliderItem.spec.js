import { mount } from '@vue/test-utils';
import PhotoSlideItem from '../PhotoSlideItem.vue';

let wrapper;
const direction = 'right';

beforeEach(() => {
  wrapper = mount(PhotoSlideItem, {
    propsData: {
      direction: direction
    }
  });
});

afterEach(() => {
  wrapper.destroy();
});

describe('PhotoSlideItem', () => {
  test('is a Vue instance', () => {
    expect(wrapper.isVueInstance).toBeTruthy();
  });
  test('should return correct props', () => {
    expect(wrapper.props().direction).toBe(direction);
  });
  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
