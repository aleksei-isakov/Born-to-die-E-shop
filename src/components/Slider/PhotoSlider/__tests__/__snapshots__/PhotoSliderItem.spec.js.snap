// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoSlideItem should match snapshot 1`] = `<transition-stub name="slide-right"></transition-stub>`;
